src/sage/graphs/domination.py

```diff
@@ -963,28 +986,24 @@ def tree_search(H, plng, dom, i):
                 raise ValueError(f"vertex ({u}) is not a vertex of the graph")
         vertices_to_dominate = {vertex_to_int[u] for u in to_dominate}
 
-    if k < 0:
-        raise ValueError("the domination distance must be a non-negative integer")
-    elif not k:
-        yield set(int_to_vertex) if to_dominate is None else set(to_dominate)
+    if not vertices_to_dominate:
+        # base case: vertices_to_dominate is empty
+        # the empty set/list is the only minimal DS of the empty set
+        yield set()
         return
     elif k > 1:
```

Comment:
The style police will complain that there's no need to use "elif" after "return"

Reply:
I changed that.